src/commonTest/kotlin/UuidTest.kt

```diff
@@ -22,10 +19,20 @@ class UuidTest {
         }
     }
 
+    @Test
+    @Suppress("DEPRECATION")
+    fun deprecated_parses_a_UUID_from_a_string() {
```

Comment:
Thoughts on code duplication in tests? I decided to keep them as separate functions, but we could extract the test logic into a separate function, somewhat along the lines of:

```kotlin
@Test
fun deprecated_parses_a_UUID_from_a_string() {
    parses_a_UUID_from_a_string { Uuid.parse(it)!! }
}

@Test
fun fromString_parses_a_UUID_from_a_string() {
    parses_a_UUID_from_a_string { Uuid.fromString(it) }
}

private fun parses_a_UUID_from_a_string(parser: (String) -> Uuid) {
    val uuid = uuid4()
    val uuidFromStr = parser(uuid.toString())
    assertEquals(uuid, uuidFromStr)
    // double check hashcode equality, while we're here
    assertEquals(uuid.hashCode(), uuidFromStr.hashCode())
}
```

Reply:
they way you have it now seems fine to me.